.github/workflows/coverity.yml

```diff
@@ -0,0 +1,91 @@
+
+name: Coverity
+
+on:
+  schedule:
+  - cron: "0 0 * * *"
+
+permissions:
+  contents: read
+
+jobs:
+  scan:
+    runs-on: ubuntu-latest
+    if: ${{ github.repository_owner == 'FreeRDP' }}
+    steps:
+    - uses: actions/checkout@v4
+    - name: Install apt dependencies
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y \
+         libxrandr-dev \
+         libxinerama-dev \
+         libusb-1.0-0-dev \
+         xserver-xorg-dev \
+         libswscale-dev \
+         libswresample-dev \
+         libavutil-dev \
+         libavcodec-dev \
+         libcups2-dev \
+         libpulse-dev \
+         libasound2-dev \
+         libpcsclite-dev \
+         xsltproc \
+         libxcb-cursor-dev \
+         libxcursor-dev \
+         libcairo2-dev \
+         libfaac-dev \
+         libfaad-dev \
+         libjpeg-dev \
+         libgsm1-dev \
+         ninja-build \
+         libxfixes-dev \
+         libxkbcommon-dev \
+         libwayland-dev \
+         libpam0g-dev \
+         libxdamage-dev \
+         libxcb-damage0-dev \
+         ccache \
+         libxtst-dev \
+         libfuse3-dev \
+         libsystemd-dev \
+         libcairo2-dev \
+         libsoxr-dev \
+         libsdl2-dev \
+         docbook-xsl \
+         libkrb5-dev \
+         libcjson-dev \
+         libpkcs11-helper1-dev \
+         libsdl2-ttf-dev \
+         libsdl2-image-dev \
+         libwebkit2gtk-4.0-dev \
+         clang \
+         libopus-dev \
+         libwebp-dev \
+         libpng-dev \
+         libjpeg-dev \
+         liburiparser-dev
+    - name: Download Coverity build tool
+      run: |
+        wget -c -N https://scan.coverity.com/download/linux64 --post-data "token=${{ secrets.COVERITY_SCAN_TOKEN }}&project=FreeRDP" -O coverity_tool.tar.gz
+        mkdir coverity_tool
+        tar xzf coverity_tool.tar.gz --strip 1 -C coverity_tool
+    - name: Build with Coverity build tool
+      run: |
+        export PATH=`pwd`/coverity_tool/bin:$PATH
+        export CC=/usr/bin/clang
+        export CXX=/usr/bin/clang++
+        export CFLAGS="-Weverything"
+        export CXXFLAGS="-Weverything"
+        cmake -DALLOW_IN_SOURCE_BUILD=true -GNinja .
```

Comment:
don´t like in source builds, sure you can´t just add `-Bbuilddir -S.` ?

Reply:
it is done in purpose. it was the easiest way for Coverity to detect source file path, otherwise it does not split file path into absolute and relative part.

Reply:
I'll add a comment with reason described